src/main/java/org/truffleruby/RubyContext.java

```diff
@@ -412,52 +410,17 @@ private String findRubyHome() throws IOException {
             return home.getCanonicalPath();
         }
 
-        // Try to find it automatically from the location of the JAR, but this won't work from the JRuby launcher as it uses the boot classpath
-
-        if (TruffleOptions.AOT) {
-            final String executablePath = (String) Compiler.command(new Object[]{"com.oracle.svm.core.posix.GetExecutableName"});
-            final String parentDirectory = new File(executablePath).getParent();
-
-            // Root of the GraalVM distribution.
-            File candidate = Paths.get(parentDirectory, "language", "ruby").toFile();
+        if (!options.LAUNCHER.isEmpty()) {
+            final File candidate = Paths.get(options.LAUNCHER).getParent().getParent().toFile();
```

Comment:
We need to canonicalize the launcher path there if not already done before, otherwise the parent might be wrong due to symlinks.

Reply:
good catch